### PR TITLE
Add example in gatsby-site for Breadcrumbs

### PR DESCRIPTION
**Pull request by ajayns**
References #82 
Gatsby plugin now supports expansions, this PR updates the tests/gatsby-starter-default to showcase the new feature.

### CODE REVIEW

**Comment by datakurre** [comment]
@ajayns We should have another test for testing something else than home-link.

Because those actual breadcrumb links should not work yet. ids are absolute urls and they should be transformed to “slugs” to be usable.

**Comment by datakurre** [comment]
IMO. We need to add slugs in test project. They may be project dependent. Once we have best practices in test project we can add helper functions into plugin package.

**Comment by ajayns** [comment]
For generating slugs in the breadcrumbs component, I'll need to get the baseUrl in the component itself so that I can process the slugs from id yes? Is this currently possible with Gatsby?

An alternative would be to pass in the baseUrl to each node or maybe to query the ploneSite in the component, use it as the baseUrl to generate slugs from the id. @datakurre What do you think?

**Comment by datakurre** [comment]
I got a second thought. We already go through every @ id in results and turn them into _id for gatsby graphql.

I know I just said that we should do slugs only in gatsby-site project, but now it feels waste of processor time to not do it when we go through every path anyways.

So, in the code where we transform every @id to _id, let’s calculate slugs there. I’d go so far that replace id value with /path/-value usable directly as gatsby slugs, but just in case, let’s leave _id as it is and set the slug as _path?

Would that make sense?

**Comment by ajayns** [comment]
Yes I'd agree. I think that'd be a good way to go with this. I'll check what I can do here and make the changes.

**Comment by ajayns** [comment]
I've added _path to the node and used them as links for the breadcrumbs but I'm unsure on how to test them using robot, any pointers on that? Maybe I should go to a page and click one of the breadcrumb links and see if reaches the page it's supposed to. @datakurre

File: tests/gatsby-starter-default/src/components/Breadcrumbs.js [hunk 1]
```diff
@@ -9,7 +9,9 @@ export default ({ data }) => (
       </li>
       {data._components.breadcrumbs.items.map(item =>
         <li key={item._id} className={'breadcrumb-item'}>
-          {item.title}
+          <Link to={`/${item._path}`}>
```
Comment (datakurre) [inline tests/gatsby-starter-default/src/components/Breadcrumbs.js]: @ajayns Shouldn’t we make such _paths that we would not need to add extra slashes in links?

**Comment by datakurre** [comment]
@ajayns It would be a lot for this pull, but could you try to replace all uses of slugs with paths in test site? Is tests still pass, it should prove that _path works.

**Comment by datakurre** [comment]
Btw, this removal for need for slugs would simplify gatsby site creation a lot :)

**Comment by ajayns** [comment]
@datakurre Yes it does, and it makes a lot of sense to do this. I've made the necessary changes and refactored the code, let's see how the CI goes!

File: tests/gatsby-starter-default/src/components/Breadcrumbs.js (hunk 1 above)
Comment (ajayns) [reply tests/gatsby-starter-default/src/components/Breadcrumbs.js]: Yes, true.

**Comment by datakurre** [comment]
@ajayns Now that’s interesting test failure.

You should be able to reproduce it locally with

make init-backend clean test

**Comment by datakurre** [comment]
🎉

**Pull request merged**